alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/resthandler/RestIndexMonitorAction.kt

```diff
@@ -78,10 +80,23 @@ class RestIndexMonitorAction(
 ) : BaseRestHandler(settings) {
 
     private var scheduledJobIndices: ScheduledJobIndices
+
     @Volatile private var maxMonitors = ALERTING_MAX_MONITORS.get(settings)
     @Volatile private var requestTimeout = REQUEST_TIMEOUT.get(settings)
     @Volatile private var indexTimeout = INDEX_TIMEOUT.get(settings)
 
+    companion object {
+        var scheduledJobSchemaVersion: Int? = null
+        var scheduledJobIndexUpdated: Boolean = false
```

Comment:
Same here. You should add `private set` to prevent from publicly modifiable.

Reply:
`scheduledJobIndexUpdated` is used by `RestIndexDestinationAction.onUpdateMappingsResponse`. When a destination or monitor is updated, will update alerting config index and set `scheduledJobIndexUpdated = true`

If add `private set`, `RestIndexDestinationAction` can't set `scheduledJobIndexUpdated` any more

Reply:
This sounds like a logic that should live in `SchemaVersionUtils`. Something like `SchemaVersionUtils.isScheduledJobIndexUpdated()` which internally does the updating logic so it can be shared between `IndexMonitorAction` and IndexDestinationAction`.

In addition, in case the index was created with newer Schema how will this be set to `true`? In this case will we do needless update on the ScheduledJob index because `scheduledJobIndexUpdated` was initialized to `false`?

Reply:
Same as last comment, will add a new util class

Reply:
Missed some comments "in case the index was created with newer Schema how will this be set to true?", will set index updated as true if index is newly created